### PR TITLE
Add in tutorial for ConditionalGradient Optimizer 

### PR DESCRIPTION
**Pull request by pkan2**
Hello! 

I have created a file for using ConditionalGradient optimizer. Please check it out! 

Thanks!

Sincerely,

Pengyu Kan

### CODE REVIEW

**Comment by pkan2** [comment]
Hi! Can you provide some feedback on this example file?

Thanks!

Sincerely,

Pengyu Kan

File: docs/tutorials/optimizers_conditionalgradient.ipynb [hunk 1]
```diff
@@ -0,0 +1,558 @@
+{
```
Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Could you refactor this notebook to match the new <code>_template.ipynb</code>:</p><p><a href="https://github.com/tensorflow/addons/blob/master/docs/tutorials/_template.ipynb" target="_blank">https://github.com/tensorflow/addons/blob/master/docs/tutorials/_template.ipynb</a></p><p> </p><p>Should have a download notebook and tensorflow.org link</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Remove the commented out lines please</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Just <code>model</code> instead of <code>model_1</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Could you make this the function's docstring</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Make <code>history_cg</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Could you re-use the same model and just run <a href="https://www.tensorflow.org/api_docs/python/tf/keras/backend/clear_session" target="_blank">https://www.tensorflow.org/api_docs/python/tf/keras/backend/clear_session</a> at the beginning of this section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Comment should read something like <code>utilize SGD or TF optimizer</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Could you further describe why this comparison of Frobenius norm is important.</p><p> </p><p>Also, could you plot the validation losses against each other.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

**Comment by seanpmorgan** [comment]
Also could you update:
https://github.com/tensorflow/addons/blob/master/docs/tutorials/_toc.yaml

File: docs/tutorials/optimizers_conditionalgradient.ipynb (hunk 1 above)
Comment (pkan2) [reply docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>I have added the tag for download notebook and TensorFlow.org. But I have not put href link into it. I am not exact sure for the link address to put in for download and Tensorflow.org part. Sorry about this.</p><p> </p>

Comment (pkan2) [reply docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>I have faced some issue with using "tf.keras.backend.clear_session()" from this link. It seems that the variables in the model has not been cleared out and the model keeps on optimizing based on previous updated variable value, which makes the model second time with using SGD start from pretty high accuracy value. Not sure for the reason. Sorry about this. </p>

Comment (seanpmorgan) [reply docs/tutorials/optimizers_conditionalgradient.ipynb]: Hmmm not sure of the reason for that. But could you revert back to `model_1` and `model_2` then.

File: docs/tutorials/optimizers_conditionalgradient.ipynb [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
```
Comment (seanpmorgan) [inline docs/tutorials/optimizers_conditionalgradient.ipynb]: <p>Can you make the Run in Colab link to <code>https://colab.research.google.com/github/tensorflow/addons/blob/master/docs/tutorials/optimizers_conditionalgradient.ipynb</code></p><p> </p><p>The link will work after it merges.</p><p> </p><p>The Tensorflow org link can be:</p><p><code>https://www.tensorflow.org/addons/tutorials/optimizers_conditionalgradient</code></p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/590/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: docs/tutorials/optimizers_conditionalgradient.ipynb (hunk 1 above)
Comment (seanpmorgan) [reply docs/tutorials/optimizers_conditionalgradient.ipynb]: Still need to write something about this.

Comment (seanpmorgan) [reply docs/tutorials/optimizers_conditionalgradient.ipynb]: See updated comments

**Comment by pkan2** [comment]
@seanpmorgan Thank you a lot for your feedbacks!

Sincerely,

Pengyu Kan
